### PR TITLE
feat(unlock-app) - verifiers should see metadata when scanning qr code

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
On the verification page, also users with verifiers status would be able to see metadata.
To solve this issue and also avoid the double sign-message issue, this PR includes: 

- [x] use new endpoint to get metadata and avoid double message sign-in
- [x] change endpoint to include metadata also for verifiers 
- [x]  remove old way to get user metadata

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/routes/metadata.ts [hunk 1]
```diff
@@ -71,7 +64,6 @@ router.get(
 )
 
 router.get('/:address/keyHolderMetadata', MetadataController.keyHolderMetadata)
-router.get('/:address/:keyId', MetadataController.data)
```
Comment (searchableguy) [inline locksmith/src/routes/metadata.ts:74]: ditto as above.

File: locksmith/src/controllers/v2/metadataController.ts [hunk 2]
```diff
@@ -41,7 +42,7 @@ export class MetadataController {
     this.web3Service = web3Service
   }
 
-  async #isKeyOrLockOwner({
+  async #isKeyOwnerOrLockVerifier({
```
Comment (julien51) [inline locksmith/src/controllers/v2/metadataController.ts:45]: Let's move that to a metadataOperations file so we can re-use?

**Review by julien51 — COMMENTED** [review]
One quick comment about making code more re-usable and keeping things outside of the controller :)

**Review by julien51 — APPROVED** [review]

File: locksmith/src/operations/metadataOperations.ts [hunk 3]
```diff
@@ -138,3 +149,32 @@ const defaultMappings = (address: string, host: string, keyId: string) => {
   defaultResponse.description = `${defaultResponse.description} Unlock is a protocol for memberships. https://unlock-protocol.com/`
   return defaultResponse
 }
+
+export const isKeyOwnerOrLockVerifier = async ({
```
Comment (searchableguy) [inline locksmith/src/operations/metadataOperations.ts:153]: Should this not take into account lock manager as well like the original API?

Afaik, we don't automatically create lock manager as verifier on the backend.

**Review by searchableguy — CHANGES_REQUESTED** [review]

File: locksmith/src/operations/metadataOperations.ts (hunk 3 above)
Comment (searchableguy) [reply locksmith/src/operations/metadataOperations.ts:153]: I'm saying it should also allow loco manager to access the data.

Comment (kalidiagne) [reply locksmith/src/operations/metadataOperations.ts:153]: Oh, get it! 
added also check for lock manager, thk!

**Comment by searchableguy** [comment]
Need to remove changes from another PR.

**Review by searchableguy — APPROVED** [review]

**Pull request merged**